web/client/observables/wfs.js

```diff
@@ -149,6 +148,50 @@ export const getJSONFeature = (searchUrl, filterObj, options = {}) => {
         }));
     }
 
+    return {
+        data,
+        queryString
+    };
+};
+
+/**
+ * Get Features in xml format.
+ * @param {string} searchUrl URL of WFS service
+ * @param {object} filterObj FilterObject
+ * @param {*} downloadOption selected format for download
+ * @returns {Observable} a stream that emits the XML
+ */
+export const getXMLFeature = (searchUrl, filterObj, options = {}, downloadOption) => {
+
+    if (options.layer && options.layer.type === 'vector') {
+        return getFeatureUtilities(searchUrl, filterObj, options, downloadOption);
+    }
```

Comment:
Are we sure we need to repeat this control?
It's a bit strange that the getFeatureUtilities can return an object or a observable

Reply:
Both `getJSONFeature` and `getXMLFeature` return observables. However, they both have similar logic and may return the same observable if `options.layer.type === 'vector'`. To ensure optimisation, they share a `getFeatureUtilities` utility function to provide them with data they need to build their respective observables. However, in the case where. In the case where `options.layer.type === 'vector'`, `getFeatureUtilities` will return the Observable needed so that both functions can send it back to their outer epics.